java/code/src/com/redhat/rhn/manager/satellite/UpgradeCommand.java

```diff
@@ -321,4 +331,23 @@ private void refreshCustomSlsFiles() {
             log.error("Error refreshing custom SLS files. Ignoring.", e);
         }
     }
+
+    /**
+     * Regenerate pillar data for every virtualization host.
+     */
+    private void refreshVirtHostPillar() {
+        try {
+            log.warn("Refreshing virtualization hosts pillar");
+            List<MinionServer> virtHosts = MinionServerFactory.listMinions().stream()
+                    .filter(Server::hasVirtualizationEntitlement)
+                    .collect(Collectors.toList());
+            virtHosts.forEach(MinionPillarManager.INSTANCE::generatePillar);
+            List<String> minionIds = virtHosts.stream().map(MinionServer::getMinionId).collect(Collectors.toList());
+            GlobalInstanceHolder.SALT_API.refreshPillar(new MinionList(minionIds));
```

Comment:
This `refresh_pillar` is handled by the minion, but for some reason it doesn't get the updated data. Running `salt '*' saltutil.refresh_pillar` manually after the update works though. I need to investigate that before merging

Reply:
The issue is a Salt one. Hacked around it by calling `saltutil.refresh_module` in `refreshPillar`